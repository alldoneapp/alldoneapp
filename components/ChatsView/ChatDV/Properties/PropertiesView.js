import React, { useEffect, useState } from 'react'
import { StyleSheet, View } from 'react-native'
import Privacy from './Privacy'
import CreatedBy from '../../../TaskDetailedView/Properties/CreatedBy'
import Highlight from './Highlight'
import DeleteChat from './DeleteChat'
import Backend from '../../../../utils/BackendBridge'
import PropertiesHeader from './PropertiesHeader'
import { useSelector } from 'react-redux'
import FollowObject from '../../../Followers/FollowObject'
import SharedHelper from '../../../../utils/SharedHelper'
import { DV_TAB_CHAT_PROPERTIES } from '../../../../utils/TabNavigationConstants'
import URLsChats, { URL_CHAT_DETAILS_PROPERTIES } from '../../../../URLSystem/Chats/URLsChats'
import Stickyness from '../../../NotesView/NotesDV/PropertiesView/Stickyness'
import ObjectRevisionHistory from '../../../NotesView/NotesDV/PropertiesView/ObjectRevisionHistory'
import { FOLLOWER_TOPICS_TYPE } from '../../../Followers/FollowerConstants'
import ProjectHelper from '../../../SettingsView/ProjectsSettings/ProjectHelper'
import { GUIDE_MAIN_CHAT_ID } from '../../../../utils/backends/Projects/guidesFirestore'
import AssistantProperty from '../../../UIComponents/FloatModals/ChangeAssistantModal/AssistantProperty'

export default function PropertiesView({ projectId, chat }) {
    const [creator, setCreator] = useState({})
    const smallScreen = useSelector(state => state.smallScreen)
    const selectedTab = useSelector(state => state.selectedNavItem)
    const loggedUser = useSelector(state => state.loggedUser)
    const accessGranted = SharedHelper.accessGranted(loggedUser, projectId)

    useEffect(() => {
        Backend.getUserOrContactBy(projectId, chat.creatorId).then(afterCreatorFetch)
        writeBrowserURL()
    }, [])

    const afterCreatorFetch = user => {
        setCreator(user)
    }

    const writeBrowserURL = () => {
        if (selectedTab === DV_TAB_CHAT_PROPERTIES) {
            const data = { projectId, chatId: chat.id }
            URLsChats.push(URL_CHAT_DETAILS_PROPERTIES, data, projectId, chat.id)
        }
    }

    const loggedUserCanUpdateObject =
        GUIDE_MAIN_CHAT_ID !== chat.id || !ProjectHelper.checkIfLoggedUserIsNormalUserInGuide(projectId)

    return (
        <View style={{ flexDirection: 'column', marginBottom: 92 }}>
            <PropertiesHeader />

            <View style={smallScreen ? localStyles.panelsContainerMobile : localStyles.panelsContainer}>
                <View style={smallScreen ? localStyles.leftContainerMobile : localStyles.leftContainer}>
                    <CreatedBy createdDate={chat.created} creator={creator} />
                    <Highlight chat={chat} projectId={projectId} disabled={!accessGranted} />
                    <Stickyness projectId={projectId} note={chat} disabled={!accessGranted} isChat />
                </View>

                <View style={smallScreen ? localStyles.rightContainerMobile : localStyles.rightContainer}>
                    <AssistantProperty
                        projectId={projectId}
                        assistantId={chat.assistantId}
                        disabled={!accessGranted || !loggedUserCanUpdateObject}
                        objectId={chat.id}
                        objectType={'chats'}
                    />
                    <Privacy projectId={projectId} chat={chat} disabled={!accessGranted} />
                    {accessGranted && (
                        <FollowObject
                            projectId={projectId}
                            followObjectsType={FOLLOWER_TOPICS_TYPE}
                            followObjectId={chat.id}
                            loggedUserId={loggedUser.uid}
                            object={chat}
                            disabled={!accessGranted}
                        />
                    )}
                </View>
            </View>

            {accessGranted && loggedUserCanUpdateObject && (
                <View style={localStyles.deleteButtonContainer}>
                    <ObjectRevisionHistory projectId={projectId} noteId={chat.noteId} />
                    <DeleteChat projectId={projectId} chat={chat} />
                </View>
            )}
        </View>
    )
}

const localStyles = StyleSheet.create({
    panelsContainer: {
        flex: 1,
        flexDirection: 'row',
        position: 'relative',
        zIndex: 50,
    },
    panelsContainerMobile: {
        flex: 1,
        flexDirection: 'column',
        position: 'relative',
        zIndex: 50,
    },
    leftContainer: {
        flex: 1,
        flexDirection: 'column',
        paddingRight: 36,
    },
    leftContainerMobile: {
        flex: 1,
        flexDirection: 'column',
    },
    rightContainer: {
        flex: 1,
        flexDirection: 'column',
        paddingLeft: 36,
    },
    rightContainerMobile: {
        flex: 1,
        flexDirection: 'column',
    },
    deleteButtonContainer: {
        marginTop: 24,
    },
})
