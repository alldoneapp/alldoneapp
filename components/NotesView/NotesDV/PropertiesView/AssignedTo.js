import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import NoteAssignee from './NoteAssignee'
import Icon from '../../../Icon'
import styles, { colors } from '../../../styles/global'
import { translate } from '../../../../i18n/TranslationService'

export default function AssignedTo({ projectId, note, disabled }) {
    return (
        <View style={localStyles.container}>
            <View style={{ marginRight: 8 }}>
                <Icon name="user" size={24} color={colors.Text03} />
            </View>
            <Text style={[styles.subtitle2, { color: colors.Text03 }]}>{translate('Owner')}</Text>
            <View style={{ marginLeft: 'auto' }}>
                <NoteAssignee projectId={projectId} note={note} disabled={disabled} />
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        maxHeight: 56,
        minHeight: 56,
        height: 56,
        paddingLeft: 8,
        paddingVertical: 8,
        alignItems: 'center',
    },
})
