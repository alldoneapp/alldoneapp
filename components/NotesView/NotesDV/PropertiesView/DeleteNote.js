import React from 'react'
import { StyleSheet, View } from 'react-native'
import DeleteNoteButton from './DeleteNoteButton'

const DeleteNote = ({ projectId, note }) => {
    return (
        <View style={localStyles.container}>
            <View style={{ marginLeft: 'auto' }}>
                <DeleteNoteButton projectId={projectId} note={note} />
            </View>
        </View>
    )
}
export default DeleteNote

const localStyles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        paddingLeft: 11,
        paddingVertical: 8,
        alignItems: 'center',
    },
})
