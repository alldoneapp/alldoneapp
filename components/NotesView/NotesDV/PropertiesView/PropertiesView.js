import React, { useEffect, useState } from 'react'
import { StyleSheet, View } from 'react-native'
import PropertiesHeader from './PropertiesHeader'
import Privacy from './Privacy'
import { useSelector, useStore } from 'react-redux'
import Highlight from './Highlight'
import Stickyness from './Stickyness'
import CreatedBy from '../../../TaskDetailedView/Properties/CreatedBy'
import ProjectHelper from '../../../SettingsView/ProjectsSettings/ProjectHelper'
import URLsNotes, { URL_NOTE_DETAILS_PROPERTIES } from '../../../../URLSystem/Notes/URLsNotes'
import FollowObject from '../../../Followers/FollowObject'
import { FOLLOWER_NOTES_TYPE } from '../../../Followers/FollowerConstants'
import Project from '../../../TaskDetailedView/Properties/Project'
import DeleteNote from './DeleteNote'
import AssignedTo from './AssignedTo'
import SharedHelper from '../../../../utils/SharedHelper'
import { DV_TAB_NOTE_PROPERTIES } from '../../../../utils/TabNavigationConstants'
import Backend from '../../../../utils/BackendBridge'
import VersionHistory from './VersionHistory'
import SaveVersion from './SaveVersion'
import AssistantProperty from '../../../UIComponents/FloatModals/ChangeAssistantModal/AssistantProperty'
import { getUserData } from '../../../../utils/backends/Users/usersFirestore'

export default function PropertiesView({ projectId, note, project }) {
    const loggedUser = useSelector(state => state.loggedUser)
    const smallScreen = useSelector(state => state.smallScreen)
    const mobile = useSelector(state => state.smallScreenNavigation)
    const isMiddleScreen = useSelector(state => state.isMiddleScreen)
    const [creator, setCreator] = useState({})
    const selectedTab = useSelector(state => state.selectedNavItem)

    const store = useStore()
    const accessGranted = SharedHelper.accessGranted(loggedUser, projectId)

    const writeBrowserURL = () => {
        if (selectedTab === DV_TAB_NOTE_PROPERTIES) {
            const data = { note: note.id, projectId: projectId }
            URLsNotes.push(URL_NOTE_DETAILS_PROPERTIES, data, projectId, note.id, note.title)
        }
    }

    useEffect(() => {
        writeBrowserURL()
    }, [projectId])

    useEffect(() => {
        writeBrowserURL()
        getUserData(note.creatorId, false).then(afterCreatorFetch)
    }, [])

    const afterCreatorFetch = user => {
        setCreator(user)
    }

    const loggedUserIsCreator = loggedUser.uid === note.creatorId
    const loggedUserCanUpdateObject =
        !note.linkedToTemplate &&
        (loggedUserIsCreator || !ProjectHelper.checkIfLoggedUserIsNormalUserInGuide(projectId))

    const isGuide = !!project.parentTemplateId

    return (
        <View
            style={
                mobile
                    ? localStyles.containerMobile
                    : isMiddleScreen
                    ? localStyles.containerTablet
                    : localStyles.container
            }
        >
            <PropertiesHeader />

            <View style={smallScreen ? localStyles.panelsContainerMobile : localStyles.panelsContainer}>
                <View style={smallScreen ? localStyles.leftContainerMobile : localStyles.leftContainer}>
                    <AssignedTo projectId={projectId} note={note} disabled={!accessGranted || isGuide} />
                    <Project item={{ type: 'note', data: note }} project={project} disabled={!accessGranted} />
                    <Highlight
                        projectId={projectId}
                        note={note}
                        disabled={!accessGranted || !loggedUserCanUpdateObject}
                    />
                    <Stickyness
                        projectId={projectId}
                        note={note}
                        disabled={!accessGranted || !loggedUserCanUpdateObject}
                    />
                </View>

                <View style={smallScreen ? localStyles.rightContainerMobile : localStyles.rightContainer}>
                    <AssistantProperty
                        projectId={projectId}
                        assistantId={note.assistantId}
                        disabled={!accessGranted || !loggedUserCanUpdateObject}
                        objectId={note.id}
                        objectType={'notes'}
                    />
                    <Privacy
                        projectId={projectId}
                        note={note}
                        disabled={!accessGranted || !loggedUserCanUpdateObject}
                    />
                    {accessGranted && (
                        <FollowObject
                            projectId={projectId}
                            followObjectsType={FOLLOWER_NOTES_TYPE}
                            followObjectId={note.id}
                            loggedUserId={store.getState().loggedUser.uid}
                            object={note}
                            disabled={!accessGranted}
                        />
                    )}
                    <CreatedBy createdDate={note.created} creator={creator} />

                    {accessGranted && loggedUserCanUpdateObject && (
                        <View style={localStyles.deleteButtonContainer}>
                            <SaveVersion projectId={projectId} note={note} />
                            <VersionHistory projectId={projectId} note={note} />
                            <DeleteNote projectId={projectId} note={note} />
                        </View>
                    )}
                </View>
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        flexDirection: 'column',
        marginBottom: 92,
        marginHorizontal: 104,
    },
    containerMobile: {
        marginHorizontal: 0,
    },
    containerTablet: {
        marginHorizontal: 56,
    },
    panelsContainer: {
        flex: 1,
        flexDirection: 'row',
        position: 'relative',
        zIndex: 50,
    },
    panelsContainerMobile: {
        flex: 1,
        flexDirection: 'column',
        position: 'relative',
        zIndex: 50,
    },
    leftContainer: {
        flex: 1,
        flexDirection: 'column',
        paddingRight: 36,
    },
    leftContainerMobile: {
        flex: 1,
        flexDirection: 'column',
    },
    rightContainer: {
        flex: 1,
        flexDirection: 'column',
        paddingLeft: 36,
    },
    rightContainerMobile: {
        flex: 1,
        flexDirection: 'column',
    },
    deleteButtonContainer: {
        marginTop: 72,
    },
})
