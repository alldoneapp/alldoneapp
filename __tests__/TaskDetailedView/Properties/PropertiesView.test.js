/**
 * @jest-environment jsdom
 */

import React from 'react'
import PropertiesView from '../../../components/TaskDetailedView/Properties/PropertiesView'
import store from '../../../redux/store'
import { storeLoggedUserProjects, setAssignee, storeCurrentUser } from '../../../redux/actions'

import renderer from 'react-test-renderer'

describe('PropertiesView component', () => {
    const navigation = {
        getParam: (prop, fallback) => {
            return {}
        },
    }
    const task = { id: 'id1', name: 'task1', recurrence: { type: 'never' } }
    store.dispatch(storeLoggedUserProjects([{ id: 'id0', color: 'white' }]))

    describe('PropertiesView snapshot test', () => {
        xit('should render correctly', () => {
            const assignee = { displayName: 'Awesomeness Maximus', photoUrl: 'tooAwesomeForIt' }
            store.dispatch(setAssignee(assignee))
            store.dispatch(storeCurrentUser({ workflow: [] }))

            const tree = renderer.create(<PropertiesView projectId="0" task={task} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })

    describe('PropertiesView methods', () => {
        xit('should set the creator in the state', () => {
            const tree = renderer.create(<PropertiesView navigation={navigation} projectId="id0" task={task} />)
            const instance = tree.getInstance()
            const test = {
                name: 'John Horton Conway',
                desc: `Your game of life made me believe the universe is indeed a computable entity since I was 15 years old. Thank you`,
            }
            instance.afterCreatorFetch(test)
            expect(instance.state.creator).toEqual(test)
        })
    })
})
