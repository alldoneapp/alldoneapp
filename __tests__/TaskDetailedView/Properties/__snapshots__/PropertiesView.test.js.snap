// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertiesView component PropertiesView snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "marginBottom": 92,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "height": 72,
        "paddingBottom": 12,
        "paddingTop": 32,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Medium",
            "fontSize": 20,
            "lineHeight": 28,
          },
          Object {
            "color": "#04142F",
          },
        ]
      }
    >
      Properties
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "position": "relative",
        "zIndex": 50,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "paddingRight": 36,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "maxHeight": 56,
            "paddingLeft": 11,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 11,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#8A94A6",
                  "fontFamily": "alldone",
                  "fontSize": 20,
                  "opacity": 1,
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#04142F",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "letterSpacing": 0.16,
                "lineHeight": 22,
              },
              Object {
                "color": "#8A94A6",
              },
            ]
          }
        >
          Assignee
        </Text>
        <View
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderColor": "#C6CDD2",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "height": 40,
                "justifyContent": "center",
                "maxHeight": 40,
                "minHeight": 40,
                "opacity": 1,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 100,
                  "height": 24,
                  "marginRight": 10,
                  "width": 24,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "#FFFFFF",
                    "flexWrap": "nowrap",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "letterSpacing": 0.8,
                    "lineHeight": 16,
                    "margin": 0,
                    "paddingHorizontal": 8,
                    "paddingVertical": 0,
                  },
                  Object {
                    "color": "#8A94A6",
                  },
                  Object {},
                ]
              }
            >
              Awesomeness
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "maxHeight": 56,
            "paddingLeft": 11,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 11,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#8A94A6",
                  "fontFamily": "alldone",
                  "fontSize": 22,
                  "opacity": 1,
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#04142F",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "letterSpacing": 0.16,
                "lineHeight": 22,
              },
              Object {
                "color": "#8A94A6",
              },
            ]
          }
        >
          Due date
        </Text>
        <View
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderColor": "#C6CDD2",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "height": 40,
                "justifyContent": "center",
                "maxHeight": 40,
                "minHeight": 40,
                "opacity": 1,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#8A94A6",
                    "fontFamily": "alldone",
                    "fontSize": 24,
                    "opacity": 1,
                  }
                }
              >
                
              </Text>
            </View>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "#FFFFFF",
                    "flexWrap": "nowrap",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "letterSpacing": 0.8,
                    "lineHeight": 16,
                    "margin": 0,
                    "paddingHorizontal": 8,
                    "paddingVertical": 0,
                  },
                  Object {
                    "color": "#8A94A6",
                  },
                  Object {},
                ]
              }
            >
              Today
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "maxHeight": 56,
            "paddingLeft": 11,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "marginRight": 8,
              "width": 24,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#8A94A6",
                  "fontFamily": "alldone",
                  "fontSize": 22,
                  "opacity": 1,
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#04142F",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "letterSpacing": 0.16,
                "lineHeight": 22,
              },
              Object {
                "color": "#8A94A6",
              },
            ]
          }
        >
          Project
        </Text>
        <View
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderColor": "#C6CDD2",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "height": 40,
                "justifyContent": "center",
                "maxHeight": 40,
                "minHeight": 40,
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 8,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 100,
                    "height": 16,
                    "width": 16,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "paddingLeft": 36,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "maxHeight": 56,
            "paddingLeft": 11,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 11,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#8A94A6",
                  "fontFamily": "alldone",
                  "fontSize": 20,
                  "opacity": 1,
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#04142F",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "letterSpacing": 0.16,
                "lineHeight": 22,
              },
              Object {
                "color": "#8A94A6",
              },
            ]
          }
        >
          Workflow
        </Text>
        <View
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderColor": "#C6CDD2",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "height": 40,
                "justifyContent": "center",
                "maxHeight": 40,
                "minHeight": 40,
                "opacity": 1,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#8A94A6",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#8A94A6",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 10,
                  }
                }
              >
                
              </Text>
            </View>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "#FFFFFF",
                    "flexWrap": "nowrap",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "letterSpacing": 0.8,
                    "lineHeight": 16,
                    "margin": 0,
                    "paddingHorizontal": 8,
                    "paddingVertical": 0,
                  },
                  Object {
                    "color": "#8A94A6",
                  },
                  Object {},
                ]
              }
            >
              Open
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "maxHeight": 56,
            "paddingLeft": 11,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 11,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#8A94A6",
                  "fontFamily": "alldone",
                  "fontSize": 22,
                  "opacity": 1,
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#04142F",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "letterSpacing": 0.16,
                "lineHeight": 22,
              },
              Object {
                "color": "#8A94A6",
              },
            ]
          }
        >
          Recurring
        </Text>
        <View
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderColor": "#C6CDD2",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "height": 40,
                "justifyContent": "center",
                "maxHeight": 40,
                "minHeight": 40,
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 8,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#8A94A6",
                    "fontFamily": "alldone",
                    "fontSize": 24,
                    "opacity": 1,
                  }
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "maxHeight": 56,
            "paddingLeft": 11,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 11,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#8A94A6",
                  "fontFamily": "alldone",
                  "fontSize": 20,
                  "opacity": 1,
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#04142F",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "letterSpacing": 0.16,
                "lineHeight": 22,
              },
              Object {
                "color": "#8A94A6",
              },
            ]
          }
        >
          Privacy
        </Text>
        <View
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderColor": "#C6CDD2",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "height": 40,
                "justifyContent": "center",
                "maxHeight": 40,
                "minHeight": 40,
                "opacity": 1,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#8A94A6",
                    "fontFamily": "alldone",
                    "fontSize": 24,
                    "opacity": 1,
                  }
                }
              >
                
              </Text>
            </View>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "#FFFFFF",
                    "flexWrap": "nowrap",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "letterSpacing": 0.8,
                    "lineHeight": 16,
                    "margin": 0,
                    "paddingHorizontal": 8,
                    "paddingVertical": 0,
                  },
                  Object {
                    "color": "#8A94A6",
                  },
                  Object {},
                ]
              }
            >
              Public
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "maxHeight": 56,
            "paddingLeft": 11,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 11,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#8A94A6",
                  "fontFamily": "alldone",
                  "fontSize": 20,
                  "opacity": 1,
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#04142F",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "letterSpacing": 0.16,
                "lineHeight": 22,
              },
              Object {
                "color": "#8A94A6",
              },
            ]
          }
        >
          Created by
        </Text>
        <View
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "paddingLeft": 11,
                "paddingVertical": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "marginRight": 12,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "borderRadius": 100,
                    "height": 40,
                    "width": 40,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#04142F",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 16,
                      "letterSpacing": 0.32,
                      "lineHeight": 24,
                    },
                    Object {
                      "color": "#04142F",
                    },
                  ]
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#04142F",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 12,
                      "letterSpacing": 0.48,
                      "lineHeight": 20,
                    },
                    Object {
                      "color": "#8A94A6",
                    },
                  ]
                }
              >
                on Today at 6:20 AM
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 40,
            "marginTop": 32,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 11,
              "paddingVertical": 8,
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#E00000",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 14,
                  }
                }
              >
                <View>
                  <Text
                    style={
                      Object {
                        "color": "#E00000",
                        "fontFamily": "alldone",
                        "fontSize": 20,
                        "opacity": 1,
                      }
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "letterSpacing": 0.8,
                      "lineHeight": 14,
                    },
                    Object {
                      "color": "#E00000",
                    },
                  ]
                }
              >
                Delete Task
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
